cryptosolver.py

```diff
@@ -195,111 +220,77 @@ def parse_constraint_input(constraint_string):
                     candidates_by_fingerprint[fingerprint].append(word)
 file_handle.close()
 
-# calculate fingerprints and candidates
-candidates_per_scrambled_word = []
-
+# sort scrambled_words by number of candidates
+# by building tuples: [scrambled_word, number_of_candidates]
+# and sorting by second field
+sortable_tuples = []
 for word in unique_scrambled_words:
     fingerprint = compute_fingerprint(word)
     candidates = candidates_by_fingerprint[fingerprint]
-    candidates_tuple = (word, fingerprint, len(candidates))
-    candidates_per_scrambled_word.append(candidates_tuple)
+    this_tuple = (word, candidates)
+    sortable_tuples.append(this_tuple)
```

Comment:
Rather than store `len(candidates)` in the tuple I just store a reference to the entire `candidate` list, and the **sort** below operates on `len(tuple[1[)`.  This way there is no need to later use the `fingerprint` to do a dictionary lookup for the `candidate` list.